mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManager.java

```diff
@@ -35,9 +35,27 @@
  * A component to manage the states of {@link TaskExecutorState} for a given {@link ResourceClusterActor}.
  */
 public interface ExecutorStateManager {
-    void putIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state);
-    void markAvailable(TaskExecutorID taskExecutorID);
-    void markUnavailable(TaskExecutorID taskExecutorID);
+    /**
+     * Store and track the given task executor's state inside this {@link ExecutorStateManager} if there is no existing
+     * state already. Ignore the given state instance if there is already a state associated with the given ID.
+     * @param taskExecutorID TaskExecutorID
+     * @param state new task executor state
+     * @return whether the given task executor becomes available.
+     */
+    boolean onTaskExecutorStateAssigned(TaskExecutorID taskExecutorID, TaskExecutorState state);
```

Comment:
nit: onTaskExecutorAssigned

What's the use for the boolean return value in this case?

Reply:
just to be consistent as the state change triggers the TE update in the executor manager.